Allow match regex to be received string-wrapped

tomjw64 opened the pull request:
Strip a single set of quotes around the regex if it was received wrapped in a string. Will allow for usage such as :match("^[a-zA-Z]")  as well as the original usage :match(^[a-zA-Z])

tomjw64 committed:
Allow match regex to be received string-wrapped

Strip a single set of quotes around the regex if it was received wrapped in a string. Will allow for usage such as :match("^[a-zA-Z]")  as well as the original usage :match(^[a-zA-Z])

reedstrm commented on cssselect2/extensions.py:
```diff
@@ -7,6 +7,8 @@
     "Callback for match pseudoClass."
     regex = serialize(selector.arguments)
+    if regex.startswith('"') and regex.endswith('"'):
+      regex = regex[1:-1]
```
To cover both, something like this:
if regex[1] in ['"','"'] and regex[1] == regex[-1]

tomjw64 committed:
Also allow match regex to be single-quote wrapped

tomjw64 committed:
Typo: replace semicolon with single quote

tomjw64 committed:
Rewrite logic

Will now capture and trim any matching starting and ending characters within trim
